docs/guidelines/gh-actions-runners.md

```diff
@@ -0,0 +1,65 @@
+# GitHub Actions and Self-Hosted Runners Guideline
+
+The scope of this guideline is to provide generic security advice for GitHub Actions and specific security advice for using self-hosted runners.
+
+In general, our first advice to teams would be to create a **Threat Model for the Software Development Lifecycle**.
+
+## GitHub Actions in General
+
+When using GitHub Actions, it is good practice to:
+
+- Investigate and familiarize yourself with the [Security Guides](https://docs.github.com/en/actions/security-guides) provided by GitHub and take appropriate actions.
```

Comment:
Should one have an explicit comment on using `Secrets` and `Environments`, where applicable - it is covered under the Security Guide of GitHub 🤔 ?

Reply:
yes, we could, we would then refer the https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions where you put the comment, and then move the stuff on "Manage secrets securely; they should only be available to workflows and branches where needed" up to this section?

Reply:
Or you may be refering to this part @aigelonic - https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment? It was on my mind when adding this part - "- Manage secrets securely; they should only be available to workflows and branches where needed."  - but i would make it more explicit.

Reply:
Yes, that could be done.
My motivation is that those who has no knowledge on GH-Actions (e.g. only worked in Azure Pipelines) it might be a bit confusing on how things are a bit "spread around" in the configurations of a repository.

E.g. that one also need to "duplicate" secrets for dependabot (if one is using that too); like the Snyk-token if one wants to Snyk-scan dependabot PRs.
(I know, you can use the Snyk integration for dependency bumping too, but that is a different story 😺)

Reply:
I will try to clarify a bit in the secret section